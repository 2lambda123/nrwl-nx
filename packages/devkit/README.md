<p style="text-align: center;"><img src="https://raw.githubusercontent.com/nrwl/nx/master/images/nx.png" width="600" alt="Nx - Smart Monorepos · Fast CI"></p>

{{links}}

<hr>

# Nx: Smart Monorepos · Fast CI

Nx is a build system with built-in tooling and advanced CI capabilities. It helps you maintain and scale monorepos, both locally and on CI.

This package contains a set of utilities for creating Nx plugins.

{{content}}
