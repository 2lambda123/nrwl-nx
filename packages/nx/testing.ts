export {cold, hot} from 'jasmine-marbles';
export {readAll} from './src/testing-utils';
