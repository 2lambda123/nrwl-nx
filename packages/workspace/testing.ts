export { MockBuilderContext } from './src/utils/testing-utils';
