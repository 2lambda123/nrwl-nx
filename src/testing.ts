export { readAll } from './utils/testing';
export { hot, cold } from 'jasmine-marbles';
