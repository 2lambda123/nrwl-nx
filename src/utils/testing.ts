import {Observable} from 'rxjs/Observable';
import {toPromise} from 'rxjs/operator/toPromise';
import {toArray} from 'rxjs/operator/toArray';

export function readAll<T>(o: Observable<T>): Promise<T[]> {
  return toPromise.call(toArray.call(o));
}
