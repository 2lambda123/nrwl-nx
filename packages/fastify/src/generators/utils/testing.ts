import { Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

export function createTreeWithFastifyApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace();
  tree.write(
    'workspace.json',
    String.raw`
      {
        "projects": {
          "${appName}": {
            "root": "apps/api",
            "sourceRoot": "apps/api/src",
            "projectType": "application",
            "prefix": "api",
            "targets":{}
          }
        }
      }
    `
  );

  return tree;
}
