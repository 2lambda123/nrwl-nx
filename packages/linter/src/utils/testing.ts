import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import { join } from 'path';
import { Tree } from '@angular-devkit/schematics';

const migrationTestRunner = new SchematicTestRunner(
  '@nrwl/linter-migrations',
  join(__dirname, '../../migrations.json')
);

const testRunner = new SchematicTestRunner(
  '@nrwl/linter',
  join(__dirname, '../../collection.json')
);

export function runMigration<T = any>(
  migrationName: string,
  options: T,
  tree: Tree
) {
  return migrationTestRunner
    .runSchematicAsync<T>(migrationName, options, tree)
    .toPromise();
}

export function runSchematic<SchemaOptions = any>(
  schematicName: string,
  options: SchemaOptions,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}
