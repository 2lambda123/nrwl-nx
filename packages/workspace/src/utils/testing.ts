import { join } from 'path';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import type { Rule, Tree } from '@angular-devkit/schematics';

const testRunner = new SchematicTestRunner(
  '@nrwl/workspace',
  join(__dirname, '../../generators.json')
);

testRunner.registerCollection(
  '@nrwl/jest',
  join(__dirname, '../../../jest/generators.json')
);

testRunner.registerCollection(
  '@nrwl/cypress',
  join(__dirname, '../../../cypress/generators.json')
);

testRunner.registerCollection(
  '@nrwl/express',
  join(__dirname, '../../../express/generators.json')
);

testRunner.registerCollection(
  '@nrwl/react',
  join(__dirname, '../../../react/generators.json')
);

testRunner.registerCollection(
  '@nrwl/angular',
  join(__dirname, '../../../angular/generators.json')
);

testRunner.registerCollection(
  '@nrwl/next',
  join(__dirname, '../../../next/generators.json')
);

testRunner.registerCollection(
  '@nrwl/node',
  join(__dirname, '../../../node/generators.json')
);

testRunner.registerCollection(
  '@nrwl/nest',
  join(__dirname, '../../../nest/generators.json')
);

testRunner.registerCollection(
  '@nrwl/web',
  join(__dirname, '../../../web/generators.json')
);

/**
 * @deprecated This will be removed in v17. Prefer writing Nx Generators with @nrwl/devkit. Generators can be tested by simply calling them in it().
 */
export function runSchematic<T extends object = any>(
  schematicName: string,
  options: T,
  tree: Tree
) {
  return testRunner.runSchematicAsync(schematicName, options, tree).toPromise();
}

/**
 * @deprecated This will be removed in v17. Prefer writing Nx Generators with @nrwl/devkit. Generators can be tested by simply calling them in it().
 */
export function callRule(rule: Rule, tree: Tree) {
  return testRunner.callRule(rule, tree).toPromise();
}
