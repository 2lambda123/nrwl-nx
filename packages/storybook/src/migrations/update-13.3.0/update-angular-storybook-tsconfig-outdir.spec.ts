import { readJson, Tree, writeJson } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import addOutDirToAngularStorybookTsconfig from './update-angular-storybook-tsconfig-outdir';

describe('Add outdir to Storybook tsconfig when uiFramework is angular migration', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace();

    writeJson(tree, 'workspace.json', {
      projects: {
        ['home-ui-angular']: {
          projectType: 'application',
          root: 'apps/myapps/ui-angular',
          sourceRoot: 'apps/myapps/ui-angular/src',
          targets: {
            storybook: {
              builder: '@nrwl/storybook:storybook',
              options: {
                uiFramework: '@storybook/angular',
                port: 4400,
                config: {
                  configFolder: 'apps/myapps/ui-angular/.storybook',
                },
              },
            },
          },
        },
      },
    });

    writeJson(tree, 'apps/myapps/ui-angular/.storybook/tsconfig.json', {
      extends: '../tsconfig.json',
      compilerOptions: {
        emitDecoratorMetadata: true,
        // apps/myapps/ui-angular/.storybook -> should be 4 directories up to root
        outDir: '../../../../dist/out-tsc'
      },
      
      exclude: [
        '../**/*.spec.ts',        
      ],
      include: ['../src/**/*', '*.js'],
    });
  });

  it(`should add outDir compilerOption`, async () => {
    await addOutDirToAngularStorybookTsconfig(tree);

    const tsConfigContent = readJson(
      tree,
      'apps/myapps/ui-angular/.storybook/tsconfig.json'
    );
    expect(tsConfigContent).toMatchSnapshot();
  });
});
