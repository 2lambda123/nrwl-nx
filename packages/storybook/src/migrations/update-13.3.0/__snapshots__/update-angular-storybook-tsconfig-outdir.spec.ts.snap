// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add outdir to Storybook tsconfig when uiFramework is angular migration should add outDir compilerOption 1`] = `
Object {
  "compilerOptions": Object {
    "emitDecoratorMetadata": true,
    "outDir": "../../../../dist/out-tsc",
  },
  "exclude": Array [
    "../**/*.spec.ts",
  ],
  "extends": "../tsconfig.json",
  "include": Array [
    "../src/**/*",
    "*.js",
  ],
}
`;
